x/gmp/types/abi_encode.go

```diff
@@ -128,27 +128,23 @@ func decToInt(amount sdk.Dec) *big.Int {
 
 var rateFactor = sdk.NewDec(10).Power(9)
 
-// NewMediansSlice creates a slice of MedianData from slices of medians and deviations.
-func NewMediansSlice(medians oracletypes.PriceStamps, deviations oracletypes.PriceStamps) ([]MedianData, error) {
+// NewMedianData creates a MedianData object of medians and deviations and their block numbers.
+func NewMedianData(medians oracletypes.PriceStamps, deviations oracletypes.PriceStamps) (MedianData, error) {
 	if len(medians) != len(deviations) {
-		return nil, fmt.Errorf("length of medians and deviations must be equal")
+		return MedianData{}, fmt.Errorf("length of medians and deviations must be equal")
 	}
-	// First, sort them so we'll be able to find matching blockNums.
-	sortedMedians := *medians.Sort()
-	sortedDeviations := *deviations.Sort()
-
-	// Then, create the MedianData slice.
-	medianData := make([]MedianData, 0, len(medians))
-	for i, median := range sortedMedians {
-		// If the median and deviation are not from the same block, skip.
-		if median.BlockNum != sortedDeviations[i].BlockNum {
-			continue
-		}
-		medianData = append(medianData, MedianData{
-			BlockNums:  []*big.Int{big.NewInt(int64(median.BlockNum))},
-			Medians:    []*big.Int{decToInt(median.ExchangeRate.Amount)},
-			Deviations: []*big.Int{decToInt(sortedDeviations[i].ExchangeRate.Amount)},
-		})
+
+	medianData := MedianData{
```

Comment:
I don't think these need to be sorted here since they should come sorted in descending block order from `HistoricMedians` and `HistoricDeviations`